src/App.tsx

```diff
@@ -1,203 +1,335 @@
+/* eslint-disable react-hooks/rules-of-hooks */
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import * as serverManager from './api/todos';
+import { Todo } from './types/Todo';
+import classNames from 'classnames';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  if (!serverManager.USER_ID) {
     return <UserWarning />;
   }
 
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState<'all' | 'active' | 'completed'>(
+    'all',
+  );
+  const [todoTitle, setTodoTitle] = useState('');
+
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [unableToLoadTodos, setUnableToLoadTodos] = useState(false);
+  const [unableToAddTodo, setUnableToAddTodo] = useState(false);
+  const [unableToUpdateTodo, setUnableToUpdateTodo] = useState(false);
+  const [unableToDeleteTodo, setUnableToDeleteTodo] = useState(false);
+  const [hasAnyError, setHasAnyError] = useState(false);
+  const [showError, setShowError] = useState(false);
```

Comment:
in this [comment](https://github.com/mate-academy/react_todo-app-loading-todos/pull/2010/commits/03f3624c6cf43fa1d8a21c8e1fb8f5ecb184d49c#r2276465706) i suggested just add generic type for error state.
you can simplify logic and make just one state instead six states
```suggestion
   enum Error {
      None = ''.
      AddTodo = 'Unable to add Todo
      ...
   }
  ...
  const [error, setError] = useState<Error>(Error.None);
```